vmware-tanzu/nsx-operator
# Add AllowOverwrite for some clients

Some clients may operate resources created by other PI so 'X-Allow-Overwrite' should be True. Add the header for those clients.
Remove headerconfig.go since it's only used for deprecated rest connector

Test Done:
1. use user=admin to run cleanup
2. check if subnetport has been deleted which created by WCP user
